annoyed-alligators/socl_media/apps/chat/models.py

```diff
@@ -0,0 +1,36 @@
+from django.db import models
+from django.conf import settings
+from django.utils.translation import ugettext_lazy as _
+
+# Create your models here.
+
+
+class Chat(models.Model):
+    """
+    A private message from user to user
+    """
+    subject = models.CharField(_("Subject"), max_length=120, blank=True)
+    body = models.TextField(_("Body"))
+    sender = models.ForeignKey(settings.AUTH_USER_MODEL,
+                               related_name='sender_messages',
+                               verbose_name=_("Sender"),
+                               on_delete=models.CASCADE)
+    recipient = models.ForeignKey(settings.AUTH_USER_MODEL,
+                                  related_name='receiver_messages', null=True,
+                                  blank=True, verbose_name=_("Recipient"),
+                                  on_delete=models.CASCADE)
+    parent_msg = models.ForeignKey('self', related_name='next_messages',
+                                   null=True, blank=True,
+                                   verbose_name=_("Parent message"),
+                                   on_delete=models.CASCADE)
+    sent_at = models.DateTimeField(_("sent at"), null=True, blank=True)
+    read_at = models.DateTimeField(_("read at"), null=True, blank=True)
```

Comment:
I don't think for 'early internet', there was a feature of seeing when the message was read. I think you can remove this too.